Reply/app/src/main/res/layout/fragment_email.xml

```diff
@@ -16,128 +16,167 @@
+              android:text="@{email.timeSent}"
+              android:textAppearance="?attr/textAppearanceBody2"
+              app:layout_constraintStart_toEndOf="@+id/sender_profile_image_view"
+              app:layout_constraintTop_toBottomOf="@+id/email_sender_text_view"
+              tools:text="2 hr ago" />
+          <!-- TODO: Temporary hardcoded black, remove when updating to M3 Typography  -->
+          <TextView
+              android:id="@+id/email_sender_text_view"
+              android:layout_width="0dp"
+              android:layout_height="wrap_content"
+              android:layout_marginStart="@dimen/grid_1"
+              android:text="@{email.sender.fullName}"
+              android:textAppearance="?attr/textAppearanceBody2"
+              android:textColor="@android:color/black"
+              app:layout_constraintStart_toEndOf="@+id/sender_profile_image_view"
+              app:layout_constraintTop_toTopOf="parent"
+              tools:text="Ali Connors" />
+          <TextView
+              android:id="@+id/email_recipients_text_view"
+              android:layout_width="0dp"
+              android:layout_height="wrap_content"
+              android:layout_marginTop="@dimen/grid_1"
+              android:text="@{email.recipientsPreview}"
+              android:textAppearance="?attr/textAppearanceBody2"
+              app:layout_constraintStart_toStartOf="parent"
+              app:layout_constraintTop_toBottomOf="@+id/sender_profile_image_view"
+              tools:text="To me, Ziad and Lily" />
+          <TextView
+              android:id="@+id/body_text_view"
+              android:layout_width="match_parent"
+              android:layout_height="wrap_content"
+              android:layout_marginTop="@dimen/grid_4"
+              android:scrollbars="vertical"
+              android:text="@{email.body}"
+              android:textAppearance="?attr/textAppearanceBody1"
+              android:textColor="@android:color/black"
+              app:layout_constraintEnd_toEndOf="parent"
+              app:layout_constraintStart_toStartOf="parent"
+              app:layout_constraintTop_toBottomOf="@+id/email_recipients_text_view" />
+          <androidx.constraintlayout.widget.Guideline
+              android:id="@+id/guidelineLeft"
+              android:layout_width="wrap_content"
+              android:layout_height="wrap_content"
+              android:orientation="vertical"
+              app:layout_constraintGuide_percent="0.5" />
+          <androidx.recyclerview.widget.RecyclerView
+              android:id="@+id/attachment_recycler_view"
+              android:layout_width="match_parent"
+              android:layout_height="wrap_content"
+              android:layout_marginTop="@dimen/grid_3"
+              app:layout_constraintEnd_toEndOf="parent"
+              app:layout_constraintStart_toStartOf="parent"
+              app:layout_constraintTop_toBottomOf="@id/body_text_view" />
```

Comment:
I think you need to constrain the bottom of this recycler view to the top of one of the email_reply_* buttons so the buttons don't overlap the attachments. Then replace `marginTop` with `marginVertical`?

There are a lot of components in this layout that have just a `marginTop`. In general, I would try to prefer symmetrical margins (`marginVertical/Horizontal`) where possible. I think it makes it much easier to create and maintain balanced layouts. I've tried to mark them with comments.

Reply:
Alright will update only top/bottom, only start/end with vertical/horizontal respectively.